sdk/core/azure-core/src/http/response.cpp

```diff
@@ -40,18 +38,72 @@ inline bool IsStringEqualsIgnoreCase(std::string const& a, std::string const& b)
 }
 } // namespace
 
-void Response::AddHeader(std::string const& name, std::string const& value)
+void Response::AddHeader(std::string const& header)
 {
-  if (IsStringEqualsIgnoreCase("Content-Length", name))
+  // get name and value from header
+  auto start = header.begin();
+  auto end = std::find(start, header.end(), ':');
+
+  if (end == header.end())
   {
-    // whenever this header is found, we reserve the value of it to be pre-allocated to write
-    // response
-    m_bodyBuffer.reserve(std::stol(value));
+    return; // not a valid header or end of headers symbol reached
   }
```

Comment:
Should an exception be thrown on an invalid header or silently ignore it?

Reply:
https://github.com/Azure/azure-sdk-for-cpp/issues/191